airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -196,23 +190,26 @@ def path(self, **kwargs) -> str:
         return "charges"
 
 
-class CustomerBalanceTransactions(SingleEmptySliceMixin, StripeStream):
+class CustomerBalanceTransactions(StripeStream):
     """
     API docs: https://stripe.com/docs/api/customer_balance_transactions/list
     """
 
     name = "customer_balance_transactions"
 
     def path(self, stream_slice: Mapping[str, Any] = None, **kwargs):
-        customer_id = stream_slice["customer_id"]
-        return f"customers/{customer_id}/balance_transactions"
+        return f"customers/{stream_slice['id']}/balance_transactions"
 
-    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
-        customers_stream = Customers(authenticator=self.authenticator, account_id=self.account_id, start_date=self.start_date)
-        slices = customers_stream.stream_slices(sync_mode=SyncMode.full_refresh)
+    def stream_slices(
+        self, *, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        parent_stream = Customers(authenticator=self.authenticator, account_id=self.account_id, start_date=self.start_date)
+        slices = parent_stream.stream_slices(sync_mode=SyncMode.full_refresh)
         for _slice in slices:
-            for customer in customers_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=_slice):
-                yield from super().read_records(stream_slice={"customer_id": customer["id"]}, **kwargs)
+            for customer in parent_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=_slice):
+                if customer["next_invoice_sequence"] == 1 and customer["balance"] == 0:
```

Comment:
Please add a comment here about why we're skipping this

Reply:
Done